### DISCUSSION TITLE
How to fix submodule access errors

### DISCUSSION THREAD

**Discussion by alexey-igrychev** [werf/werf#6338] [General]
When using **werf**, you need to provide the necessary access credentials for werf to synchronize submodules in its service worktree.

Given the following `.gitmodules` configuration:

```
[submodule "example"]
  path = example
  url = https://github.com/org/repo.git
```

A user will encounter the following error:

```
fatal: could not read Username for 'https://github.com': No such device or address
```

To work with submodules in this case, you can consider the following options:

**1. Use a token:**

```
git config --global url."https://<TOKEN>@github.com/org/repo.git".insteadOf "https://github.com/org/repo.git"
```

**2. Use SSH:**

```
git config --global url."git@github.com:org/repo.git".insteadOf "https://github.com/org/repo.git"
```

If the submodule is located in the same hosting platform (e.g., GitHub or GitLab) as the main project, it’s more appropriate to use relative paths:

```
[submodule "example"]
  path = example
  url = ../repo.git
```

**Comment by wowsel** [comment]
yeah. It would be great to add some section with faq about submodules to the documentation, at least with this command.
`git config --global url."https://<TOKEN>@github.com/org/repo.git".insteadOf "https://github.com/org/repo.git"`
